applications/minotari_console_wallet/src/init/mod.rs

```diff
@@ -804,6 +810,53 @@ pub(crate) fn boot_with_password(
     Ok((boot_mode, password))
 }
 
+pub fn prompt_wallet_type(
+    boot_mode: &WalletBoot,
+    wallet_config: &WalletConfig,
+    non_interactive: &bool,
+) -> Option<WalletType> {
+    if *non_interactive {
+        return Some(WalletType::Software);
+    }
+
+    if wallet_config.wallet_type.is_some() {
+        return wallet_config.wallet_type;
+    }
+
+    match boot_mode {
+        WalletBoot::New => {
+            if prompt("\r\nWould you like to use a connected hardware wallet? (Supported types: Ledger)") {
+                print!("Scanning for connected Ledger hardware device... ");
+                let err = "No connected device was found. Please make sure the device is plugged in before continuing.";
+                match TransportNativeHID::new(&HidApi::new().expect(err)) {
+                    Ok(_) => {
+                        println!("Device found.");
+                        let account = prompt_ledger_account().expect("An account value");
+                        Some(WalletType::Ledger(account))
```

Comment:
The `TRANSPORT`  lazy static should be assigned a value
```suggestion
                    Ok(transport) => {
                        *TRANSPORT.lock().expect("lock exists") = Some(transport);
                        println!("Device found.");
                        let account = prompt_ledger_account().expect("An account value");
                        Some(WalletType::Ledger(account))
```

Reply:
This actually sent me on a wild goose chase trying to find where the code had previously been removed.

But, we don't actually need to assign this here. We check to make sure the ledger is present to see if we should create a ledger backed wallet. But the ledger could be unplugged anytime between this prompt, and performing a transaction. It doesn't need to be plugged in at all times. We need to check to see if it's present again, only during script and meta data sig generation for send transactions.